nemo/collections/nlp/models/question_answering/qa_model.py

```diff
@@ -159,11 +183,81 @@ def validation_epoch_end(self, outputs):
                 do_lower_case=self.do_lower_case,
             )
 
+            if self.validation_config.output_nbest_file is not None:
```

Comment:
> There are various issues right now. Common themes are -
> 
> * Way too verbose configs. Users will need several lines to override each tokenizer from its defaults
How do you assess verbosity? length of the config part? Or or there specific parameters that are redundant in your view?
> * Attaching `bert_` name to generic components like tokenizers
we do that in our old code, huggingface does this. And this will not be changed in this PR
> * ASR BPE tests need to be fixed.
done

Reply:
Good point, whats the effect on OmegaConf.to_container if you send a None into it?
> it breaks

Reply:
I dont think thats a good idea, HF can change its API in the future and break a lot of code. Its best to have a think wrapper around this, so that we can change just the wrapper when necessary instead of dozens of underlying models which directly use the code.

Think of the TokenizerSpec like a thin adapter between our NeMo models and their HF tokenizers.
> i removed inheritance, so this issue is outdated now

Reply:
As to reducing verbosity - does every parameter need to be in the config - say do_lower_case? What parameters can you leave out and have good defaults instead? If it cant be reduced, thats fine, but having to add 5 lines of overrides to switch from WPE to SPE is going to get painful fast for the user. For reference, ASR basically limit this by explicitly requesting tokenizer path and tokenizer type. Thats it. All other decisions are left to the code (which I can see why NLP model cannot allow, user needs the flexibility to edit all those values too). 

For now, leave it as is I guess.

> we do that in our old code, huggingface does this. 
You mean to say HF *only ever uses* bert tokenizer? Thats news to me.

> it breaks 
Ok so lets check for null before we pass it into that function? One of the configs does have it as null by default with commented out config settings.

Reply:
No, i mean HF also initializes tokenizer with "bert-base-cased", so using pretrained_model_name